Task 10: Filter articles by topic

sausagroll90 opened the pull request:
Users can use a drop down menu to filter articles by topic.

pippaamy commented on __tests__/app.test.js:
```diff
@@ -54,35 +54,85 @@ describe("/api/topics", () => {
+        });
+    });
+
+    test("200: topic query: should only list articles wih given topic", () => {
```
spell error here

pippaamy commented on __tests__/app.test.js:
```diff
@@ -54,35 +54,85 @@ describe("/api/topics", () => {
+          expect(msg).toBe("topic not found");
+        });
+    });
+
```
really thorough testing, well done!

pippaamy reviewed (commented):
Great job, this is really well tested!!!

sausagroll90 committed:
Feature: Add filtering by topic to homepage

Previously the home page was a list of all articles.
Now users can use a drop down menu to filter articles by topic.

pippaamy reviewed (commented):
Love that it shows in the url, this is working really well - well done!
